advocacy_docs/edb-postgres-ai/migration-etl/data-migration-service/getting_started/preparing_db/preparing_oracle_source_databases.mdx

```diff
@@ -0,0 +1,283 @@
+
+```sql
+SELECT GROUP#, TYPE, MEMBER FROM V_$LOGFILE;
+
+    GROUP# TYPE    MEMBER
+---------- ------- --------------------------------------------------
+         1 ONLINE  /opt/oracle/oradata/ORCLCDB/redo03.log
+         2 ONLINE  /opt/oracle/oradata/ORCLCDB/redo01.log
+         3 ONLINE  /opt/oracle/oradata/ORCLCDB/redo04.log
+
+SELECT GROUP#, ARCHIVED, BYTES/1024/1024 MB, STATUS FROM V_$LOG;
+
+    GROUP# ARC  MB STATUS
+---------- --- --- ----------------
+         1 YES 2000 INACTIVE
+         3 YES 2000 INACTIVE
+         3 NO  2000 CURRENT
+```
+
+This example uses three log groups of size 2000MB. Each group has one file member. This might be too
+small for many production databases. You can safely adjust the redo logs with synchronous commands such as the following:
+
+```sql
+  ALTER DATABASE ADD LOGFILE GROUP 4 ('/opt/oracle/oradata/ORCLCDB/redo04.log') SIZE 8G;
+  ALTER DATABASE ADD LOGFILE GROUP 5 ('/opt/oracle/oradata/ORCLCDB/redo05.log') SIZE 8G;
+  ALTER DATABASE ADD LOGFILE GROUP 6 ('/opt/oracle/oradata/ORCLCDB/redo06.log') SIZE 8G;
+  ALTER DATABASE ADD LOGFILE GROUP 7 ('/opt/oracle/oradata/ORCLCDB/redo07.log') SIZE 8G;
+  ALTER SYSTEM ARCHIVE LOG CURRENT;
+  ALTER SYSTEM CHECKPOINT;
+  ALTER SYSTEM ARCHIVE LOG CURRENT;
+  ALTER SYSTEM CHECKPOINT;
+  ALTER SYSTEM ARCHIVE LOG CURRENT;
+  ALTER SYSTEM CHECKPOINT;
+  ALTER DATABASE DROP LOGFILE GROUP 1;
+  ALTER DATABASE DROP LOGFILE GROUP 2;
+  ALTER DATABASE DROP LOGFILE GROUP 3;
+```
+
+These commands result in four new 8GB log groups. Each group has a single log file.
+
+Consult your DBA for appropriate production sizing.
+
+### Create a user with limited privileges for data migration
+
+#### Tablespace preparation
+
+Provide a database user with adequate roles to carry out the CDC process.
+
+Then, we recommend creating a tablespace for the CDC user. For container databases, you need to create a pluggable database as well.
+
+This example creates a tablespace and datafiles for CDC migration. Your database settings might vary, but a common configuration with `SMALLFILE` tablespaces and an 8kB database block size results in a maximum of 32GB of storage avaiable per `MAXSIZE` tablespace datafile. Therefore, you might need to add multiple `AUTOEXTEND` datafiles when this limit might be exceeded.
+
+```sql
+-- Create the tablespace, or in the case of a CDB/PDB, create the CDB tablespace
+CREATE TABLESPACE <TABLESPACE_NAME> DATAFILE '/opt/oracle/oradata/ORCLCDB/logminer_tbs.dbf' SIZE 25M REUSE AUTOEXTEND ON MAXSIZE UNLIMITED;
+
+-- For CDB/PDB deployments we must specify at least one tablespace datafile for the PDB
+CREATE TABLESPACE <TABLESPACE_NAME> DATAFILE '/opt/oracle/oradata/ORCLCDB/ORCPDB1/logminer_tbs_1.dbf' SIZE 25M REUSE AUTOEXTEND ON MAXSIZE UNLIMITED;
+-- Additional data files can be added with as follows
+ALTER TABLESPACE <TABLESPACE_NAME> DATAFILE '/opt/oracle/oradata/ORCLCDB/ORCLPDB1/logminer_tbs_2.dbf' SIZE 25M REUSE AUTOEXTEND ON MAXSIZE UNLIMITED;
```

Comment:
This `ALTER TABLESPACE` statement seems to be incorrect. It is missing the `ADD` keyword before the `DATAFILE` keyword.  The correct version of the command is:
> 
ALTER TABLESPACE <TABLESPACE_NAME> ADD DATAFILE '/opt/oracle/oradata/ORCLCDB/ORCLPDB1/logminer_tbs_2.dbf' SIZE 25M REUSE AUTOEXTEND ON MAXSIZE UNLIMITED;

Reply:
got it.